src/test/codegen/align-byval.rs

```diff
@@ -0,0 +1,20 @@
+// ignore-x86
+//
+// Tests that `byval` alignment is properly specified (#80127).
+// Note that 32-bit x86 has special rules (see #103830), and it's therefore
+// ignored here.
+
+#[repr(C)]
+#[repr(align(16))]
+struct Foo {
+    a: [i32; 16],
+}
+
+extern "C" {
+    // CHECK: declare void @f({{.*}}byval(%Foo) align 16{{.*}})
+    fn f(foo: Foo);
```

Comment:
You have to have a certain number of byval parameters for the original bug to surface. Enough to fill up the registers. I think the function signature in #80127 was the minimal repro; I remember trying to minimize it further and having not much luck.

Reply:
If it's a regression test it should never start to succeed, it could only ever start to fail right? I guess I'm confused by what you're saying

Reply:
Idk, I think I fundamentally don't trust a test that just checks the LLVM IR, because I don't understand it well enough for that to be convincing over a real integration test that proves that the C code can operate with the Rust code. But I recognize that that's probably quite hard to test in the rust repo. Perhaps someone else knows a way?

Reply:
> If it's a regression test it should never start to succeed, it could only ever start to fail right? I guess I'm confused by what you're saying.

I'm saying it could get broken accidentally but we wouldn't notice it because it could accidentally start to pass. I can add an interop test though.

Reply:
That would be awesome. Thank you for tackling this bug that has existed for so so long 🙏🙏

Reply:
Added the test.